modules/google-maps/src/google-maps-overlay.js

```diff
@@ -114,11 +114,25 @@ export default class GoogleMapsOverlay {
     const _customRender = () => {
       this._overlay.requestRedraw();
     };
-    this._deck = createDeckInstance(this._map, this._overlay, this._deck, {
+    const deck = createDeckInstance(this._map, this._overlay, this._deck, {
       gl,
       _customRender,
       ...this.props
     });
+    this._deck = deck;
+
+    // By default, animationLoop._renderFrame invokes
+    // animationLoop.onRender. We override this to
+    // postpone the invocation until we are within
+    // the _onDrawVector lifecycle function, to avoid
+    // changing GL state when the Google Maps library
+    // isn't expecting it
+    deck.animationLoop._renderFrame = () => {
+      deck._pickAndCallback();
+      if (deck.layerManager.needsUpdate()) {
```

Comment:
There are other reasons deck may want to rerender, like effects or layerFilter prop change.